dspace-api/src/main/java/org/dspace/harvest/OAIHarvester.java

```diff
@@ -900,94 +902,46 @@ protected void alertAdmin(int status, Exception ex) {
         String oaiSetId = harvestRow.getOaiSetId();
         String metaPrefix = harvestRow.getHarvestMetadataConfig();
 
-        return verifyOAIharvester(oaiSource, oaiSetId, metaPrefix, true);
+        return harvestedCollectionService.verifyOAIharvester(oaiSource, oaiSetId, metaPrefix, true);
     }
 
     /**
-     * Verify the existence of an OAI server with the specified set and
-     * supporting the provided metadata formats.
+     * Return all available metadata formats
      *
-     * @param oaiSource  the address of the OAI-PMH provider
-     * @param oaiSetId   OAI set identifier
-     * @param metaPrefix OAI metadataPrefix
-     * @param testORE    whether the method should also check the PMH provider for ORE support
-     * @return list of errors encountered during verification. Empty list indicates a "success" condition.
+     * @return a list containing a map for each supported  metadata format
      */
-    public static List<String> verifyOAIharvester(String oaiSource,
-                                                  String oaiSetId, String metaPrefix, boolean testORE) {
-        List<String> errorSet = new ArrayList<String>();
-
-        // First, see if we can contact the target server at all.
-        try {
-            new Identify(oaiSource);
-        } catch (Exception ex) {
-            errorSet.add(OAI_ADDRESS_ERROR + ": OAI server could not be reached.");
-            return errorSet;
-        }
+    public static List<Map<String,String>> getAvailableMetadataFormats() {
+        List<Map<String,String>> configs = new ArrayList<>();
+        String metaString = "oai.harvester.metadataformats.";
+        Enumeration pe = Collections.enumeration(
+            DSpaceServicesFactory.getInstance().getConfigurationService().getPropertyKeys("oai")
+        );
+        while (pe.hasMoreElements()) {
```

Comment:
You don't need to enumerate all the keys one by one in a Collection...you should be able to use `getPropertyKeys("oai.harvester.metadataformats");` to get a List of all matching keys, instead of looping through everything starting with "oai" and then looking for "oai.harvester.metadataformats" within that.